### ISSUE TITLE
How to add localhost to Allowed Domains in Client Security during testing?

### ISSUE THREAD

**Issue by pratik60** [getsentry/sentry#1461]
I tried this but anything from localhost:3000 is not working. Any ideas on what I could do?

```
http://localhost*
http://127.0.0.1*
```

My temporary solution was using localtunnel but that's not ideal during development, right?

**Comment by mattrobenolt** [comment]
You can just use `*` to catch anything. 

--
Matt Robenolt
@mattrobenolt




On Tue, Mar 17, 2015 at 4:43 AM -0700, "Pratik Bothra" <notifications@github.com> wrote:












I tried this but anything from localhost:3000 is not working. Any ideas on what I could do?

http://localhost*
http://127.0.0.1*




My temporary solution was using localtunnel but that's not ideal during development, right?

**Comment by sometimescasey** [comment]
Hey, sorry for revisiting an old thread, and I might be missing something super basic here - but even with Allowed Domains set to just `*` I still get the following message ("Cannot fetch resource due to restricted IP address") when looking at an error from a React app at localhost:3000:

![image](https://user-images.githubusercontent.com/16248601/94817325-e67a8b80-03ca-11eb-86ca-60b200404757.png)

Any way to resolve? Thank you!

**Comment by dcramer** [comment]
Unless this is entirely an internal instance what you're wanting to do is unsafe - we block local network access by default, so even if you're self-hosting Sentry and this access pattern is valid, its still not desirable.

https://docs.sentry.io/platforms/javascript/sourcemaps/uploading/

**Comment by sometimescasey** [comment]
Hi - thank you so much for the quick reply!

Would you be able to ELI5 for a beginner why this is unsafe and undesirable? I was following this tutorial which uses Sentry for error reporting (https://serverless-stack.com/chapters/setup-an-error-boundary-in-react.html) and one of the steps is to deliberately cause a local error and see it show up in Sentry.

I can see why this wouldn't be the intended usage (no need for error logging during local development, really) but would love to understand your comment above. Ty!

**Comment by dcramer** [comment]
@sometimescasey basically if you allow code to hit a local network connection - and that code is at all controlled by a third party (in this case it would be, as your end-users could submit a fake error with fake source maps/etc) - you then open up a vector for them doing things they shouldn't do.

A likely not possible, but good example is if you ran something like a Redis server, and the user was able to craft a payload that caused Sentry to run a "flushdb" command on that Redis server. We disallow any local network access because otherwise they can access _any_ network service running on local host. If nothing else they can force a basic HTTP request to it, which could have unknown consequences.

**Comment by sometimescasey** [comment]
That makes sense - thanks so much for taking the time to explain.
